### PR TITLE
configure: HDF4 requires built-in pds format

### PR DESCRIPTION
**Pull request by tbonfort**
Allow configure to be run with --with-hdf4 and --disable-all-optional-drivers

### CODE REVIEW

**Comment by rouault** [comment]
This is acceptable, but a perhaps better alternative would be to move frmts/pds/nasakeywordhandler.[cpp/h] into gcore/ ?

**Comment by tbonfort** [comment]
I'd like to leave out driver specific code from gcore for the sake of having minimal builds stay small and build fast (I know this particular case is not that significant, but every little helps). Should I introduce some driver specific ifdefs in gcore/nasakeywordhandler.* to only build  if needed ?

**Comment by rouault** [comment]
> Should I introduce some driver specific ifdefs in gcore/nasakeywordhandler.* to only build if needed ?

That wouldn't be immediate as the logic to append -DFRMT_xxxx to the compilation line is in frmts/GNUmakefile and frmts/makefile.vc, not gcore/

Let's go with your initial commit.

**Comment by tbonfort** [comment]
> That wouldn't be immediate as the logic to append -DFRMT_xxxx to the compilation line is in frmts/GNUmakefile and frmts/makefile.vc, not gcore/
I can easily add the defines to the gcore/GNUMakefile . makefile.vc being hardcoded can also be trivially updated. Let me know

**Comment by rouault** [comment]
> Let me know

up to you. I'd just prefer if we could avoid to have a long line of -DFRMT_ passed to every .c/.cpp files built in gcore/

File: gdal/gcore/GNUmakefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ OBJ	=	gdalmultidim.o gdalopeninfo.o gdaldrivermanager.o gdaldriver.o gdaldataset
 		gdalpython.o gdalpythondriverloader.o tilematrixset.o \
 		gdal_thread_pool.o
 
-CPPFLAGS	:=	 -I../frmts/gtiff -I../frmts/mem -I../frmts/vrt -I../ogr -I../ogr/ogrsf_frmts/generic -I../gnm/ -I../gnm/gnm_frmts/ $(JSON_INCLUDE) -I../ogr/ogrsf_frmts/geojson $(CPPFLAGS) $(PAM_SETTING) $(XTRA_OPT)
+CPPFLAGS	:=	 -I../frmts/gtiff -I../frmts/mem -I../frmts/vrt -I../ogr -I../ogr/ogrsf_frmts/json -I../ogr/ogrsf_frmts/generic -I../gnm/ -I../gnm/gnm_frmts/ $(JSON_INCLUDE) -I../ogr/ogrsf_frmts/geojson $(CPPFLAGS) $(PAM_SETTING) $(XTRA_OPT)
```
Comment (rouault) [inline gdal/gcore/GNUmakefile:20]: there's no ogr/ogrsf_frmts/json .

**Comment by tbonfort** [comment]
The tests are failing due to -Werror and -Weffc++ being used in gcore:
```
In file included from nasakeywordhandler.cpp:55:0:
nasakeywordhandler.h:48:15: error: ‘class NASAKeywordHandler’ has pointer data members [-Werror=effc++]
 class CPL_DLL NASAKeywordHandler
               ^~~~~~~~~~~~~~~~~~
nasakeywordhandler.h:48:15: error:   but does not override ‘NASAKeywordHandler(const NASAKeywordHandler&)’ [-Werror=effc++]
nasakeywordhandler.h:48:15: error:   or ‘operator=(const NASAKeywordHandler&)’ [-Werror=effc++]
nasakeywordhandler.cpp: In constructor ‘NASAKeywordHandler::NASAKeywordHandler()’:
nasakeywordhandler.cpp:71:1: error: ‘NASAKeywordHandler::osHeaderText’ should be initialized in the member initialization list [-Werror=effc++]
 NASAKeywordHandler::NASAKeywordHandler() :
 ^~~~~~~~~~~~~~~~~~
nasakeywordhandler.cpp:71:1: error: ‘NASAKeywordHandler::oJSon’ should be initialized in the member initialization list [-Werror=effc++]
cc1plus: all warnings being treated as errors
../GDALmake.opt:661: recipe for target 'nasakeywordhandler.o' failed
make[1]: *** [nasakeywordhandler.o] Error 1
```
I don't have the time to go down this rabbithole for now, I thought this would just be a simple dependency pr/issue.

**Comment by rouault** [comment]
@tbonfort -Weffc++ should hopefully be adressed by my follow-up commits in https://github.com/OSGeo/gdal/pull/2740

**Pull request closed without merging**